zeppelin-interpreter/src/test/java/org/apache/zeppelin/interpreter/remote/AppendOutputRunnerTest.java

```diff
@@ -0,0 +1,181 @@
+     * It has been observed that for 10,000 continuos event
+     * calls, 30-40 Web-socket calls are made. Keeping
+     * the unit-test to a pessimistic 100 web-socket calls.
+     */
+    verify(listener, atMost(NUM_CLUBBED_EVENTS)).onOutputAppend(any(String.class), any(String.class), any(String.class));
+    CheckAppendOutputRunner.stopRunnerForUnitTests();
+  }
+
+  @Test
+  public void testWarnLoggerForLargeData() throws InterruptedException {
+    String data = "data\n";
+    int numEvents = 100000;
+    for (int i=0; i<numEvents; i++) {
+      AppendOutputRunner.appendBuffer("noteId", "paraId", data);
+    }
+
+    TestAppender appender = new TestAppender();
+    Logger logger = Logger.getRootLogger();
+    logger.addAppender(appender);
+    Logger.getLogger(RemoteInterpreterEventPoller.class);
+
+    RemoteInterpreterProcessListener listener = mock(RemoteInterpreterProcessListener.class);
+    loopForCompletingEvents(listener, 1);
+    List<LoggingEvent> log;
+    do {
+      log = appender.getLog();
+    } while(log.size() != 2);
+    LoggingEvent sizeWarnLogEntry = null;
+
+    for (LoggingEvent logEntry: log) {
+      if (Level.WARN.equals(logEntry.getLevel())) {
+        sizeWarnLogEntry = logEntry;
+      }
+    }
+    String loggerString = "Processing size for buffered append-output is high: " +
+        (data.length() * numEvents) + " characters.";
+    assertTrue(loggerString.equals(sizeWarnLogEntry.getMessage()));
+    CheckAppendOutputRunner.stopRunnerForUnitTests();
+  }
+
+  private class BombardEvents implements Runnable {
+
+    @Override
+    public void run() {
+      String noteId = "noteId";
+      String paraId = "paraId";
+      for (int i=0; i<NUM_EVENTS; i++) {
+        AppendOutputRunner.appendBuffer(noteId, paraId, "data\n");
+      }
+    }
+  }
+
+  private class TestAppender extends AppenderSkeleton {
+    private final List<LoggingEvent> log = new ArrayList<LoggingEvent>();
+
+    @Override
+    public boolean requiresLayout() {
+        return false;
+    }
+
+    @Override
+    protected void append(final LoggingEvent loggingEvent) {
+        log.add(loggingEvent);
+    }
+
+    @Override
+    public void close() {
+    }
+
+    public List<LoggingEvent> getLog() {
+        return new ArrayList<LoggingEvent>(log);
+    }
+  }
+
+  private void prepareInvocationCounts(RemoteInterpreterProcessListener listener) {
+    doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws Throwable {
+        numInvocations += 1;
+        return null;
+      }
+    }).when(listener).onOutputAppend(any(String.class), any(String.class), any(String.class));
+  }
+
+  private void loopForCompletingEvents(RemoteInterpreterProcessListener listener, int numTimes) {
+    numInvocations = 0;
+    prepareInvocationCounts(listener);
+    AppendOutputRunner.setListener(listener);
+    CheckAppendOutputRunner.startRunnerForUnitTests();
+    while(numInvocations != numTimes);
```

Comment:
I suggest you add `Circuit Breaker` with few seconds to avoid running infinitely.